### ISSUE TITLE
Improvement Idea: Auto builds / CD pipe

### ISSUE THREAD

**Issue by ksaadDE** [xou816/spot#697]
What about regular autobuilds and publishing to flatpak?  At best we also get a release file with the exec for all supported systems inkl. flatpak here. :)

If that's possible in your time sched and if it's maintainable. It would also make the install instructions and dependencies more fixed an clear, as it kinda establishes a CD pipe 

@Diegovsky

**Comment by Diegovsky** [comment]
Yeah, I would be down to do it. Problem is, this takes a long time to run on actions, #685 .

I don't have experience with github actions or flatpak builder (trying to build a snapshot of spot right at this moment), but I think it's an awesome idea.

If you have the experience, do send a PR to my fork!

**Comment by ksaadDE** [comment]
> takes a long time to run on actions

After I looked at [spot's current Flatpak workflow](https://github.com/xou816/spot/blob/development/.github/workflows/spot-snapshots.yml), it might be suitable to use caching for the long-living dependencies

How Workflow Caching works is seemingly documented [here](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows)

Someone else did that for Rust on [StackOverFlow (here)](https://stackoverflow.com/a/64635207) as well

> I don't have experience with github actions

I had no time yet to extensively work with Github's Workflows either.

> do send a PR to my fork!

If I see something that can be improved and when I am having the time I will surely file one

**Comment by Diegovsky** [comment]
Thank you very much for your contributions @ksaadDE !

**Comment by Diegovsky** [comment]
hey, if you're still interested: open a PR right here in this repo. It would help a lot!

**Comment by ksaadDE** [comment]
> hey, if you're still interested

Interested yes but hella busy. I'm sorry!

**Comment by Diegovsky** [comment]
Dw! I'm unfortunately busy too
